src/Normalize.ml

```diff
@@ -222,3 +222,101 @@ let equal_use_file ~ignore_doc_comments ast1 ast2 =
+                    ; pexp_loc
+                    ; pexp_attributes }
+                  , [] )
+            ; pstr_loc } ] ) ->
+        (* normalize consecutive whitespace chars to a single space *)
+        let doc' =
+          String.concat ~sep:" "
+            (List.filter ~f:(Fn.non String.is_empty)
+               (String.split_on_chars doc
+                  ~on:['\t'; '\n'; '\011'; '\012'; '\r'; ' ']))
+        in
+        let attr =
+          ( {txt; loc}
+          , m.payload m
+              (PStr
+                 [ { pstr_desc=
+                       Pstr_eval
+                         ( { pexp_desc=
+                               Pexp_constant (Pconst_string (doc', None))
+                           ; pexp_loc
+                           ; pexp_attributes }
+                         , [] )
+                   ; pstr_loc } ]) )
+        in
+        docstrings := (loc, doc') :: !docstrings ;
+        attr
+    | attr -> Ast_mapper.default_mapper.attribute m attr
+  in
+  (* sort attributes *)
+  let attributes (m : Ast_mapper.mapper) atrs =
+    let atrs = List.filter atrs ~f:doc_attribute in
+    Ast_mapper.default_mapper.attributes m
+      (List.sort ~compare:Poly.compare atrs)
+  in
+  {Ast_mapper.default_mapper with attribute; attributes}
+
+let docstrings_impl s =
+  let docstrings = ref [] in
+  let _ : structure = map_structure (make_docstring_mapper docstrings) s in
+  !docstrings
+
+let docstrings_intf s =
+  let docstrings = ref [] in
+  let _ : signature = map_signature (make_docstring_mapper docstrings) s in
+  !docstrings
+
+let docstrings_use_file s =
+  let docstrings = ref [] in
+  let _ : toplevel_phrase list =
+    map_use_file (make_docstring_mapper docstrings) s
+  in
+  !docstrings
+
+let moved_docstrings get_docstrings s1 s2 =
+  let d1 = get_docstrings s1 in
+  let d2 = get_docstrings s2 in
+  let equal (_, s1) (_, s2) = String.equal s1 s2 in
+  match List.zip d1 d2 with
+  | None ->
+      (* We only return the ones that are not in both lists. *)
+      (* [l1] contains the ones that disappeared. *)
+      let l1 = List.filter d1 ~f:(fun x -> not (List.mem ~equal d2 x)) in
+      let l1 = List.map ~f:(fun (l, s) -> (l, Location.none, s)) l1 in
+      (* [l2] contains the ones that appeared. *)
+      let l2 = List.filter d2 ~f:(fun x -> not (List.mem ~equal d1 x)) in
+      let l2 = List.map ~f:(fun (l, s) -> (Location.none, l, s)) l2 in
+      List.rev_append l1 l2
+  | Some l ->
+      (* We suppose that no docstring appeared nor disappeared, meaning that
+         each docstring is present in both lists but may have different
+         locations. *)
+      let different ((_, s1), (_, s2)) = not (String.equal s1 s2) in
+      let l = List.filter l ~f:different in
+      let d1, d2 = List.unzip l in
+      List.map d1 ~f:(fun (l, s) ->
+          let str_equal (_, s') = String.equal s s' in
+          let l' = List.find_exn d2 ~f:str_equal |> fst in
```

Comment:
@gpetiot this is raising in combination with #460 on e.g. test/code/infer/infer/src/checkers/PulseDomain.ml . Any ideas why?

Reply:
There are several other failures when running `make -C test test_setup test_unstage test_clean test_pull test` from that PR branch:
```
30% 224:502=4s code/infer/infer/src/IR/Subtype.ml                                                        ocamlformat.exe: Cannot process "code/infer/infer/src/IR/Exp.ml".
  Please report this bug at https://github.com/ocaml-ppx/ocamlformat/issues.
  BUG: unhandled exception. Use [--debug] for details.
46% 336:390=4s code/infer/infer/src/biabduction/BuiltinDefn.ml                                           ocamlformat.exe: Cannot process "code/infer/infer/src/biabduction/Match.ml".
  Please report this bug at https://github.com/ocaml-ppx/ocamlformat/issues.
  BUG: comments changed.
48% 351:375=4s code/infer/infer/src/base/ProcessPool.ml                                                  ocamlformat.exe: Cannot process "code/infer/infer/src/biabduction/Tabulation.ml".
  Please report this bug at https://github.com/ocaml-ppx/ocamlformat/issues.
  BUG: unhandled exception. Use [--debug] for details.
52% 384:342=4s code/infer/infer/src/quandary/QuandarySummary.ml                                          ocamlformat.exe: Cannot process "code/infer/infer/src/biabduction/Prop.ml".
  Please report this bug at https://github.com/ocaml-ppx/ocamlformat/issues.
  BUG: unhandled exception. Use [--debug] for details.
63% 463:263=3s code/infer/infer/src/clang/objcProperty_decl.mli                                          ocamlformat.exe: Cannot process "code/infer/infer/src/clang/cTL.mli".
  Please report this bug at https://github.com/ocaml-ppx/ocamlformat/issues.
  BUG: unhandled exception. Use [--debug] for details.
69% 503:223=2s code/infer/infer/src/checkers/cost.mli                                                    ocamlformat.exe: Cannot process "code/infer/infer/src/checkers/accessPathDomains.mli".
  Please report this bug at https://github.com/ocaml-ppx/ocamlformat/issues.
  BUG: unhandled exception. Use [--debug] for details.
86% 631:95=1s code/infer/infer/src/concurrency/StarvationModels.mli                                      ocamlformat.exe: Cannot process "code/infer/infer/src/concurrency/RacerDDomain.mli".
  Please report this bug at https://github.com/ocaml-ppx/ocamlformat/issues.
  BUG: unhandled exception. Use [--debug] for details.
96% 699:27=0s code/infer/infer/src/base/Language.mli                                                     ocamlformat.exe: Cannot process "code/infer/infer/src/biabduction/Prop.mli".
  Please report this bug at https://github.com/ocaml-ppx/ocamlformat/issues.
  BUG: unhandled exception. Use [--debug] for details.
100% 726:0=0s code/infer/infer/src/quandary/QuandarySummary.mli
make[1]: *** [test_inplace] Error 8
```